client/src/components/menu-views/new-menu-item-dish.js

```diff
@@ -0,0 +1,313 @@
+        }
+      });
+  }
+
+  render() {
+    const {
+      categories, units, selectedCategory, selectedPortionSize, isShowAlert, errors, portionPrice, portionSize,
+    } = this.state;
+    return (
+      <Container fluid className="new-menu-item-container">
+        <h5>
+          Add a new dish
+        </h5>
+        <AlertBase
+          show={isShowAlert}
+          type="danger"
+          title={Object.values(errors).join('')}
+        />
+        <Form.Group>
+          <Form.Label>Select category: </Form.Label>
+          <br />
+          <Dropdown>
+            <Dropdown.Toggle>{selectedCategory}</Dropdown.Toggle>
+            <Dropdown.Menu>
+              {categories.map((category) => (
+                <Dropdown.Item
+                  onSelect={(e) => { this.setState({ selectedCategory: e, selectedCategoryId: category.id }); }}
+                  eventKey={category.name}
+                  key={category.id}
+                >
+                  {category.name}
+                </Dropdown.Item>
+              ))}
+            </Dropdown.Menu>
+          </Dropdown>
+        </Form.Group>
+
+        <Form.Group>
+          <Form.Label>Dish name</Form.Label>
+          <Form.Control
+            type="text"
+            name="dishName"
+            placeholder="Please enter the dish name"
+            onChange={(e) => this.handleChange(e)}
+          />
+        </Form.Group>
+
+        <Form.Group>
+          <Form.Label>Ingredients: (e.g.: chicken, ham, cheese)</Form.Label>
+          <Form.Control
+            type="text"
+            name="ingredients"
+            placeholder="Please enter the ingredient list"
+            onChange={(e) => this.handleChange(e)}
+          />
+        </Form.Group>
+
+        <Form.Group>
+          <Form.Label>Portion size: </Form.Label>
+          <Form.Row>
+            <Col>
+              <Form.Control
+                type="number"
+                placeholder="Please enter portion size"
+                name="portionSize"
+                value={portionSize}
+                onChange={(e) => this.handleChange(e)}
+              />
+            </Col>
+            <Col>
+              <View
+                className="portion-size"
+                id="portion-size"
+                name={selectedPortionSize}
+                values={units}
+                onSelect={(e) => { this.setState({ selectedPortionSize: e }); }}
+              />
+            </Col>
+          </Form.Row>
+        </Form.Group>
+
+        <Form.Group>
+          <Form.Label>Portion price: </Form.Label>
+          <Form.Row>
+            <Col>
+              <Form.Control
+                type="number"
+                name="portionPrice"
+                value={portionPrice}
+                onChange={(e) => this.handleChange(e)}
+                placeholder="Price"
+              />
+            </Col>
+            <Col className="col">
+              <span className="">UAN</span>
+            </Col>
```

Comment:
You should handle errors from server (e.g. server can be down or returns error)
And our server doesn't save invalid price (more than 2 digits after comma), but your form lets to send this invalid price
![image](https://user-images.githubusercontent.com/6152066/81061559-39fcb200-8edd-11ea-8daf-9bc33b4dac20.png)

Reply:
Fixed